openpype/hosts/resolve/api/lib.py

```diff
@@ -202,22 +212,39 @@ def create_media_pool_item(fpath: str,
     if existing_mpi:
         return existing_mpi
 
-    dirname, file = os.path.split(fpath)
-    _name, ext = os.path.splitext(file)
+    files = []
+    first_frame = frame_start - handle_start
+    last_frame = frame_end + handle_end
+    dir_path = os.path.dirname(fpath)
+    base_name = os.path.basename(fpath)
 
-    # add all data in folder to mediapool
-    media_pool_items = media_storage.AddItemListToMediaPool(
-        os.path.normpath(dirname))
+    # prepare glob pattern for searching
+    padding = len(str(last_frame))
+    str_first_frame = str(first_frame).zfill(padding)
 
-    if not media_pool_items:
-        return False
+    # convert str_first_frame to glob pattern
+    # replace all digits with `?` and all other chars with `[char]`
+    # example: `0001` -> `????`
+    glob_pattern = re.sub(r"\d", "?", str_first_frame)
+
+    # in filename replace number with glob pattern
+    # example: `filename.0001.exr` -> `filename.????.exr`
+    base_name = re.sub(str_first_frame, glob_pattern, base_name)
+
+    # get all files in folder
+    for file in glob.glob(os.path.join(dir_path, base_name)):
+        files.append(file)
+
+    # iterate all files and check if they exists
+    # if not then remove them from list
+    for file in files[:]:
+        if not os.path.exists(file):
+            files.remove(file)
```

Comment:
This is a perfect example of a function that we should be abstracting out of here.

A `get_representation_files(representation)` or whatever so that we can accurately get the sequence we need instead of implementing the file logic into this loader itself.

Reply:
Are you talking about pipeline wide abstraction? This is abstracted to `resolve.api.lib`

Reply:
I did mean that, yes. ❤️ 

But didn't see this was in resolve lib already. In that case - can you just still separate out the function that retrieves the paths themselves, so that `create_media_pool_item()`  just calls that function.

I can imagine there are reasons to retrieve the expected filepaths for a representation (which is what this is doing!) without directly calling `ImportMedia` ;)

Reply:
Just wanted to cross-reference this issue since it's _somewhat_ related: https://github.com/ynput/OpenPype/issues/4361

Reply:
implemented via [`0a71b89` (#5673)](https://github.com/ynput/OpenPype/pull/5673/commits/0a71b89ddd857676d1561c3cdbfeb690ebae6103)